physionet-django/training/models.py

```diff
@@ -0,0 +1,142 @@
+from django.db import models
+
+from project.modelcomponents.fields import SafeHTMLField
+
+
+class OnPlatformTraining(models.Model):
+    training_type = models.ForeignKey('user.TrainingType',
+                                      on_delete=models.CASCADE, related_name='op_trainings')
+    version = models.FloatField(default=1.0)
+
+    class Meta:
+        default_permissions = ('change',)
+        unique_together = ('training_type', 'version')
+
+    def __str__(self):
+        return f'{self.training_type} v{self.version}'
+
+
+class Module(models.Model):
+    name = models.CharField(max_length=100)
+    training = models.ForeignKey('training.OnPlatformTraining',
+                                 on_delete=models.CASCADE, related_name='modules')
+    order = models.PositiveIntegerField()
+    description = SafeHTMLField()
+
+    class Meta:
+        unique_together = ('training', 'order')
+
+    def __str__(self):
+        return self.name
+
+
+class Quiz(models.Model):
+    question = SafeHTMLField()
+    module = models.ForeignKey('training.Module',
+                               on_delete=models.CASCADE, related_name='quizzes')
+    order = models.PositiveIntegerField()
+
+
+class ContentBlock(models.Model):
+    module = models.ForeignKey('training.Module',
+                               on_delete=models.CASCADE, related_name='contents')
+    body = SafeHTMLField()
+    order = models.PositiveIntegerField()
+
+
+class QuizChoice(models.Model):
+    quiz = models.ForeignKey('training.Quiz',
+                             on_delete=models.CASCADE, related_name='choices')
+    body = models.TextField()
+    is_correct = models.BooleanField('Correct Choice?', default=False)
+
+
+class OnPlatformTrainingProgress(models.Model):
+    class Status(models.TextChoices):
+        IN_PROGRESS = 'IP', 'In Progress'
+        COMPLETED = 'C', 'Completed'
+
+    user = models.ForeignKey('user.User', on_delete=models.CASCADE)
+    training = models.ForeignKey('training.OnPlatformTraining', on_delete=models.CASCADE)
+    status = models.CharField(max_length=2, choices=Status.choices, default=Status.IN_PROGRESS)
+    started_datetime = models.DateTimeField(auto_now_add=True)
```

Comment:
In the comment, I think you mentioned that we don't want to capture the start time, or did I misunderstand? Anyway, I do think we should capture the start time, because it is interesting/helpful to know how long it takes people to complete the course.

Reply:
Sorry for the confusion, We do capture the start time for a course and each module inside the course, but we donot track start time for a quiz(question) and a content block. my idea was that the quiz, content block would probably just take few minutes to complete, so i was not sure if its a good idea to track the start time.

Reply:
Got it, thanks, makes sense.